Adds advanced settings descriptions (#6928)
opensearch-project/documentation-website: _dashboards/management/advanced-settings.md

{::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/icons/alert-icon.png" class="inline-icon" alt="alert icon"/>{:/} **Note**<br>Certain settings require you to modify [the `opensearch_dashboards.yml` file](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/config/opensearch_dashboards.yml) and restart OpenSearch Dashboards.
{: .note}

## Required permissions

To modify settings, you must have permission to make changes. See [Multi-tenancy configuration](https://opensearch.org/docs/latest/security/multi-tenancy/multi-tenancy-config/#give-roles-access-to-tenants) for guidance about assigning role access to tenants.

## Advanced settings descriptions

The following tables describe the core advanced settings.

### General settings

The following table describes the **General** settings:

Setting | Description
:--- | :---

> `csv:quoteValues`  | Defines whether to Enclose values containing special characters or multiline values with a double quotation mark `"`. Default is `On`.  |

Reviewer: enclose

> `csv:quoteValues`  | Defines whether to Enclose values containing special characters or multiline values with a double quotation mark `"`. Default is `On`.  |

Author: ```suggestion
`csv:quoteValues`  | Defines whether to enclose values containing special characters or multiline values with a double quotation mark `"`. Default is `On`.  |
```

`csv:separator`  | Defines whether to use a specific character or string to delimit exported values. Default is `,`.  |
`dateFormat`  | Defines the format for displaying dates. Default is `MMM D, YYYY @ HH:mm:ss.SSS`.  |
`dateFormat:dow`  | Defines the day on which to start the week. Default is `Sunday`.  |
`dateFormat:scaled`  | Defines the format for timestamps. The timestamps format changes depending on the time between measurements (hour, minutes, seconds, and milliseconds). The keys are time periods in [ISO8601](https://www.iso.org/iso-8601-date-and-time-format.html) format: `YYYY-MM-DD`.   |
`dateFormat:tz`  | Defines the time zone for OpenSearch Dashboards. Default is the time zone detected by your browser.  |
`dateNanosFormat`  | Defines the format for representing dates with nanoseconds. Default is `MMM D, YYYY @ HH:mm:ss.SSSSSSSSS`.  |

> `defaultIndex`  | Defines the default index for all indexes in an OpenSearch cluster. Default is `null`.  |

Reviewer: Should we say that `If there are no indices added, defaultIndex will be set to null by default; otherwise, it will be set to the first added index`?

Author: @abbyhu2000 Revised. Please review and make any necessary changes.

`defaultRoute`  | Defines the gateway point. Use this setting to change the landing page for OpenSearch Dashboards. The setting must be a relative URL. Default is `/app/home`. |
`fields:popularLimit` | Defines the number of N fields to be displayed. Default is `10`.  |
`filterEditor:suggestValues` | Defines whether the filter editor suggests field values. Default is `Off`.  |

> `filters:pinnedByDefault`  | Defines whether filters are automatically pinned. Default is `Off.`  |

Reviewer: We can add `each filter can be set to be pinned by clicking the filter and selecting 'Pin across all apps'`?

Author: @abbyhu2000 Change made. Please review and make any necessary changes.

`format:number:defaultLocale`  | Defines the language locale for numerals. Default is `en`.  |
`format:number:defaultPattern`  | Defines the default numeral format for the number format. Default is `0,0.[000]`.  |
`format:percent:defaultPattern`  | Defines the default numeral format for the percent format. Default is `0,0.[000]%`.  |
`histogram:barTarget`  |  Defines a specified number of bars for date histograms that use the `auto` interval. Default is `50`.  |
`histogram:maxBars`  | Defines the maximum number of bars to show in date histograms. Default is `100`.  |
`history:limit` | Defines how many of the most recent values to store in the history. Default is `10`.  |
`indexPattern:placeholder`  | Defines the placeholder value to use when creating index patterns.  |
`metaFields` | Enables fields that are not part of the `_source` field to be merged into the document. Default is `_source`, `_id`, `_type`, `_index`, and `_score`.  |
`metrics:max_buckets` | Defines the maximum number of buckets that a single data source can return. Default is `2000`.  |
`query:allowLeadingWildcards`  | Defines whether `*` is allowed as the first character in a query clause. Default is `On`.  | 

> `query:queryString:options`  |  Defines the options for the Lucene query string parser. Default is `true`.  |

Reviewer: Default Lucene query string parser is 

`{ "analyze_wildcard": true }`

Author: Revised

> `reporting:useFOR`  | Enables or disables `ForeignObject` rendering for embedding of external content into reports. Default is `On`.  |

Reviewer: This two `reporting:useFOR` and  `reporting:useOcr` will only be shown after installing reporting plugin. Should we make a note here? since the default advanced setting page will not have these setting.

Author: @abbyhu2000 REvised. Please review and make any necessary changes.

> `reporting:useOcr`  | Enables or disables optical character recognition (OCR) to be run on PRF reports. The `reporting:useFOR` and `reporting:useOcr` options are visible only after installing the reporting plugin. Default is `Off`.  |

Author: Good catch. It should be PDF.

> `reporting:useOcr`  | Enables or disables optical character recognition (OCR) on PRF reports. The `reporting:useFOR` and `reporting:useOcr` options are visible only after installing the reporting plugin. Default is `Off`.  |

Author: ```suggestion
`reporting:useOcr`  | Enables or disables optical character recognition (OCR) on PDF reports. The `reporting:useFOR` and `reporting:useOcr` options are visible only after installing the reporting plugin. Default is `Off`.  |
```

`savedObjects:listingLimit`  | Defines the number of objects to fetch when viewing a listing page. Default is `1000`.  |
`savedObjects:perPage`  | Defines the number of objects to display on each page of the load dialog. Default is `20`.  |
`search:queryLanguage`  | Defines the query language for OpenSearch Dashboards. Default is `DQL`.  |
`shortDots:enable`  | Enables or disables the shortening of long fields. Default is `Off`.  |
`sort:options`  | Defines the options for the sort parameter. Default is `boolean`.  |

> `state:storeInSessionStorage`  | Enables or disables the storing of URLs. Default is `Off`.  |

Reviewer: The definitions here are not accurate: URLs are always stored. The default is off which means the URLs will not be stored in session storage, but will be stored in browsers.

Author: @abbyhu2000 Revised. Please review and make any necessary changes.

`context:step`  | Defines the number by which to increment or decrement the context size. Default is `5`.  |
`context:tieBreakerFields`  | Defines the fields that are used to break a tie between documents that have the same timestamp value. The first field that is present and sortable in the current index pattern is used. Default is `_doc`.  |
`defaultColumns`  | Defines the columns that appear by default on the **Discover** page. Default is `_source`.  |
`discover:aggs:terms:size`  | Defines the number of terms that are visualized when selecting the **Visualize** button in the field dropdown. Default is `20`.  |
`discover:modifyColumnsOnSwitch`  | Defines whether unavailable columns are removed from a new index pattern. Default is `On`.  |
`discover:sampleSize`  | Defines the number of rows to show in a table. Default is `20`.  |
`discover:searchOnPageLoad`  | Defines whether a search is run when **Discover** first loads. The setting does not affect the loading of saved searches. Default is `On`.  |
`discover:sort:defaultOrder`  | Defines how to sort time-based index patterns. Default is `Descending`.  |

> `doc_table:hideTimeColumn`  | Defines whether to hide the `Time` column in the **Discover** application and in all saved searches on dashboards. Default is `Off`.  |

Author: ```suggestion
`doc_table:hideTimeColumn`  | Defines whether to hide the `Time` column in the **Discover** application and in all saved dashboard searches. Default is `Off`.  |
```

`notifications:lifetime:warning`  | Defines the duration for which warning notifications are displayed. Default is `10000` milliseconds.Set field to `Infinity` to disable notifications.

## Search settings

The following table describes the **Search** settings:

Setting | Description
:--- | :--- 
`courier:batchSearches`  | Enables or disables how dashboard panels load. When disabled, panels load individually, and search requests end when users navigate away or update the query. When enabled, all panels load together when all data is loaded, and searches do not end. Default is `Off`.  |
`courier:customRequestPreference`  | Enables or disables whether to use the [request preference]({{site.url}}{{site.baseurl}}//api-reference/popular-api/) when setting is set to `custom`. Default is `_local`.  |
`courier:ignoreFilterIfFieldNotInIndex`  | Enables or disables support for dashboards that contain visualizations using different indexes. When disabled, all filters are applied to all visualizations. When enabled, filters are ignored for a visualization if the visualization's index does not contain the field being filtered. Default is `Off`.  |

> `courier:maxConcurrentShardRequests`  | Defines the maximum number of concurrent shard requests that can be made for `_msearch` requests sent by OpenSearch Dashboards. Set it to `0` to disable the setting and to use the default value set by OpenSearch. Default is `0`.  |

Author: revised to "...that OpenSearch Dashboards can initiate for `_msearch` requests."

`visualization:dimmingOpacity`  | Defines the opacity of chart items that are dimmed when another chart element is highlighted. The lower the value, the more the highlighted element will stand out. The value must be between `0` and `1`. Default is `0.5`.  |
`visualization:enablePluginAugmentation`  | Enables or disables access to plugin functionality through line chart visualizations. Default is `On`.  |
` line chart visualizations`  | Defines the maximum number of associated augmentations per visualization. Default is `10`. Associating more than 10 plugin resources per visualization can cause performance issues.  |
`visualization:heatmap:maxBuckets`  | Defines the maximum number of buckets that a single data source can return in a heat map visualization. A higher number of buckets can negatively impact browser rendering performance. Default is `50`. |
`visualization:regionmap:customVectorMapMaxSize`  | Defines the maximum number of features that can be loaded from a custom vector map. Default is `1000`.  |
`visualization:regionmap:showWarnings`  | Specifies whether a warning is shown when terms cannot be joined to a shape on a region map. Default is `On`.  |

> `visualization:tileMap:WMSdefaults`  | Defines the default [properties](https://leafletjs.com/reference.html#tilelayer-wms) for the WMS map server in coordinate maps. Default is `enabled: false`.  |

Author: ```suggestion
`visualization:tileMap:WMSdefaults`  | Defines the default [properties](https://leafletjs.com/reference.html#tilelayer-wms) for the Web Map Service (WMS) map server in coordinate maps. Default is `enabled: false`.  |
```
